### ISSUE TITLE
pod install fails on RN 0.63

### ISSUE THREAD

**Issue by maxgr0** [mrousavy/react-native-blurhash#58]
# Bug

**Describe the bug**
When trying to install this module, the installation via `pod install` fails with the following reasons:
```
[!] The following Swift pods cannot yet be integrated as static libraries:

The Swift pod `react-native-blurhash` depends upon `React-Core`, which does not define modules. To opt into those targets generating module maps (which is necessary to import them from Swift when building as static libraries), you may set `use_modular_headers!` globally in your Podfile, or specify `:modular_headers => true` for particular dependencies.
```
Tried to set `use_modular_headers!` on top of my Podfile but this produces a mass of other errors. Any help would be highly appreciated.

# Environment
- CocoaPods 1.9.1
- RN 0.63

**Additional context**
```json
"react": 16.13.1,
"react-native": 0.63.2,
"react-native-blurhash": 1.0.24
```

**Comment by dburdan** [comment]
I ran into the same problem. Seems to be an upstream issue with React-Core, possibly due to changes in Xcode 14. I was able to successfully build for now with the following patch:

`react-native+0.63.2.patch`
```patch
diff --git a/node_modules/react-native/React-Core.podspec b/node_modules/react-native/React-Core.podspec
index cf65263..4252543 100644
--- a/node_modules/react-native/React-Core.podspec
+++ b/node_modules/react-native/React-Core.podspec
@@ -49,7 +49,10 @@ Pod::Spec.new do |s|
   s.header_dir             = "React"
   s.framework              = "JavaScriptCore"
   s.library                = "stdc++"
-  s.pod_target_xcconfig    = { "HEADER_SEARCH_PATHS" => "\"$(PODS_TARGET_SRCROOT)/ReactCommon\" \"$(PODS_ROOT)/boost-for-react-native\" \"$(PODS_ROOT)/DoubleConversion\" \"$(PODS_ROOT)/Folly\"" }
+  s.pod_target_xcconfig    = {
+    "HEADER_SEARCH_PATHS" => "\"$(PODS_TARGET_SRCROOT)/ReactCommon\" \"$(PODS_ROOT)/boost-for-react-native\" \"$(PODS_ROOT)/DoubleConversion\" \"$(PODS_ROOT)/Folly\"",
+    "DEFINES_MODULE" => "YES"
+  }
   s.user_target_xcconfig   = { "HEADER_SEARCH_PATHS" => "\"$(PODS_ROOT)/Headers/Private/React-Core\""}
   s.default_subspec        = "Default"
```

**Comment by maxgr0** [comment]
> I ran into the same problem. Seems to be an upstream issue with React-Core, possibly due to changes in Xcode 14. I was able to successfully build for now with the following patch:
> 
> `react-native+0.63.2.patch`
> 
> ```diff
> diff --git a/node_modules/react-native/React-Core.podspec b/node_modules/react-native/React-Core.podspec
> index cf65263..4252543 100644
> --- a/node_modules/react-native/React-Core.podspec
> +++ b/node_modules/react-native/React-Core.podspec
> @@ -49,7 +49,10 @@ Pod::Spec.new do |s|
>    s.header_dir             = "React"
>    s.framework              = "JavaScriptCore"
>    s.library                = "stdc++"
> -  s.pod_target_xcconfig    = { "HEADER_SEARCH_PATHS" => "\"$(PODS_TARGET_SRCROOT)/ReactCommon\" \"$(PODS_ROOT)/boost-for-react-native\" \"$(PODS_ROOT)/DoubleConversion\" \"$(PODS_ROOT)/Folly\"" }
> +  s.pod_target_xcconfig    = {
> +    "HEADER_SEARCH_PATHS" => "\"$(PODS_TARGET_SRCROOT)/ReactCommon\" \"$(PODS_ROOT)/boost-for-react-native\" \"$(PODS_ROOT)/DoubleConversion\" \"$(PODS_ROOT)/Folly\"",
> +    "DEFINES_MODULE" => "YES"
> +  }
>    s.user_target_xcconfig   = { "HEADER_SEARCH_PATHS" => "\"$(PODS_ROOT)/Headers/Private/React-Core\""}
>    s.default_subspec        = "Default"
> ```

thanks for this @dburdan! 
@mrousavy is there any hint where I could start investigating this bug on plugin-side to avoid waiting for an official RN patch?

**Comment by mrousavy** [comment]
Hi! Thanks for the patch @dburdan, hopefully this will land in RN soon.

As for the problem, I found out that when using Xcode 12 the build failed (see: https://github.com/facebook/react-native/issues/29633#issuecomment-694187116)

I've changed the dependencies, but that fix only worked for Objective-C and since rn-blurhash is a Swift module I probably have to revert https://github.com/mrousavy/react-native-blurhash/commit/76930054363dd185814aae90aa1a5932aef9c297 and https://github.com/mrousavy/react-native-blurhash/commit/adc1cf0c6f702c6d8c3099433678ae4f8fc7523d for now - at least until an official fix/support comes from the React-Core pod.

**Comment by maxgr0** [comment]
that would be really awesome @mrousavy! Switching back to `1.0.23` works for me with the setup mentioned above. BTW: Great library! 💫

**Comment by mrousavy** [comment]
I've reverted the breaking change and published `1.0.26`. That should work for now, if anyone experiences build errors similar to https://github.com/facebook/react-native/issues/29633 please let me know thanks 👍

**Comment by mrousavy** [comment]
0.63.3 landed with a fix for that (https://github.com/facebook/react-native/commit/e4ec71a1d9674991560bb4519d909c113c6e86f4), will update the podspec soon, so all versions from now on only work with >0.63.3 (or the patch from @dburdan)

Closing this
